### ISSUE TITLE
placement of "new" comments form

### ISSUE THREAD

**Issue by frase-git** [cmb69/twocents_xh#57]
Hoffentlich kein Duplikat (#10)

Ich bin gerade bei einer Seite mit 2Cents und wünsche mir den Button "Neuen Kommentar verfassen" **_unterhalb_** der angezeigten Kommentare.

Ginge das als Konfig-Option?

Im Moment behelfe ich mir mit einer Änderung der Reihenfolge in "views/comments.php" - was aber nicht updatesicher ist.

**Comment by cmb69** [comment]
> Ich bin gerade bei einer Seite mit 2Cents und wünsche mir den Button "Neuen Kommentar verfassen" unterhalb der angezeigten Kommentare.

Derzeit sollte es so sein, dass dieser Schalter im Prinzip (Paginierung außen vor) dort angezeigt, wo dann auch der gepostete Kommentar steht (eben abhängig von `comments_order`). Passt das nicht?

**Comment by frase-git** [comment]
Version: 1.0beta3
(Keine Paginierung)
Egal, ob ASC oder DESC - der Button wird immer _oberhalb_ der Kommentare angezeigt.

**Comment by cmb69** [comment]
Hm, bei mir (auch 1.0beta3) klappt es wie es soll. Zur Erklärung: der Button selbst wird per JS erzeugt, und sollte genau an der Stelle angezeigt werden, an der auch das Formular ausgegeben wurde. Also bitte mal temporär JS deaktivieren, und prüfen, ob das Formular richtig platziert ist. Falls nicht, dann liegt es vielleicht an einem modifizierten `views/comments.php` (die [entsprechende Zuweisung im Controller](https://github.com/cmb69/twocents_xh/blob/1.0beta3/classes/MainController.php#L160-L161) sieht zumindest korrekt aus). Falls das Formular selbst korrekt platziert ist, dann gibt es wohl ein JS Problem, und man müsste browserseitig weiter suchen.

**Comment by frase-git** [comment]
Verstehe.
Habe mit DESC angefangen und es stand **oben**.
Dann habe ich die comments.php manipuliert.
Habe also nicht richtig "geguckt".

Das Problem bleibt, dass es nicht konfigurierbar ist.
Ich möchte DESC haben und den Button (das Formular) unten.
Das geht momentan nur mit Manipulation.

**Comment by frase-git** [comment]
Übrigens:
Ich habe einen Text über den Kommentaren, in dem auf die Kommentarmöglichkeit unten hingewiesen wird. Interner Link zum Formular.
(War eine alte Seite mit "comments". Da hatte das Formular eine ID, die man anspringen konnte.)
das geht mit Twocents gar nicht mehr - oder? (manipulieren)

Und jetzt habe ich noch ein Problem im Formular (HTML):
"I" = italic lässt sich nicht mehr ausschalten.
Die Knopf-Formatierung "inset/outset" spinnt auch.
Kannst du das nachvollziehen?

**Comment by cmb69** [comment]
> Ich möchte DESC haben und den Button (das Formular) unten.

Bedeutet halt eine zusätzliche Konfigoption, aber okay.

> Ich habe einen Text über den Kommentaren, in dem auf die Kommentarmöglichkeit unten hingewiesen wird. Interner Link zum Formular.
(War eine alte Seite mit "comments". Da hatte das Formular eine ID, die man anspringen konnte.)
das geht mit Twocents gar nicht mehr - oder? (manipulieren)

Das geht leider tatsächlich nicht. Ich muss allerdings vermutlich [diesbezüglich sowieso etwas umbauen](https://github.com/cmb69/twocents_xh/issues/58) (das Formular per JS zu verstecken, war wohl nicht die beste Idee). Wenn das so geändert wurde, dann könntest du direkt auf dieses Formular verlinken.

> Der Button braucht eine ID - nicht das Formular, da es zunächst unsichtbar ist.

Kommt halt drauf an, ob das JS ausgeführt wird, oder nicht. Und dem Button eine ID zu verpassen, wird nicht unbedingt funktionieren, da dieser u.U. zu spät erzeugt wird, und der Browser bereits vorher versucht hat, den Anker anzuspringen.

> Und jetzt habe ich noch ein Problem im Formular (HTML):
"I" = italic lässt sich nicht mehr ausschalten.
Die Knopf-Formatierung "inset/outset" spinnt auch.
Kannst du das nachvollziehen?

Das passiert wohl nur, wenn unterschiedliche Schriftarten für italic/bold definiert wurden. Da erzeugt der Browser noch zusätzlich ein entsprechendes `<font>` Element, entfernt es aber nicht wieder. Probier im Zweifel mal das Kommentarformular mit einer gemeinsamen Schriftart für roman/italic/bold zu stylen. Ich schau aber mal, ob ich das nicht allgemein in den Griff bekomme.

**Comment by frase-git** [comment]
> Der Button braucht eine ID - nicht das Formular, da es zunächst unsichtbar ist.
>> Kommt halt drauf an, ob das JS ausgeführt wird, oder nicht. Und dem Button eine ID zu verpassen, wird nicht unbedingt funktionieren, da dieser u.U. zu spät erzeugt wird, und der Browser bereits vorher versucht hat, den Anker anzuspringen.

Stimmt. Bei ausgeschaltetem JS klappt eines nicht, bei eingeschaltetem das andere. :(

> Das passiert wohl nur, wenn unterschiedliche Schriftarten für italic/bold definiert wurden.

Autsch! An was man alles denken muss.
Test:
```css
.twocents_editor {
    font-family: arial,sans-serif !important;
	font-weight: normal;
	font-style: normal;
...
}
.twocents_editor b {
    font-family: arial,sans-serif !important;
    font-weight: bold;
    font-style: normal;
}
.twocents_editor i {
    font-family: arial,sans-serif !important;
    font-weight: normal;
    font-style: italic;
}
```
Das klappt soweit, aber was wird mit Kombinationen aus `b i` ?

**Comment by cmb69** [comment]
> Das klappt soweit, aber was wird mit Kombinationen aus b i ?

Siehe #59 (TL;DR: ich kann vermutlich fixen, aber das müsste serverseitig noch korrigiert werden). Bitte weitere Kommentare zu diesem Thema besser dort vornehmen, damit man auch später noch einen Überblick haben kann. :)

> Es sind die gleichen Schriftarten, aber unterschiedliche Schriftschnitte.

Danke für die Erklärung (ich versuche mir "Schriftschnitte" zu merken). :)

> Bei CSS lädt man allerdings tatsächlich unterschiedliche Dateien, sodass hier unterschiedliche Schriftarten nicht falsch ist.

Und im Extremfall sogar richtig (z.B. `* {font-family:serif} i {font-family:sans-serif}`), auch wenn das typografisch wohl arger Blödsinn wäre.
